model/host/db.go

```diff
@@ -567,6 +567,20 @@ func FindOneId(id string) (*Host, error) {
 	return FindOne(ById(id))
 }
 
+func FindOneByIdOrTag(id string) (*Host, error) {
+	query := db.Query(bson.M{
+		"$or": []bson.M{
+			bson.M{TagKey: id},
+			bson.M{IdKey: id},
+		},
+	})
+	host, err := FindOne(query) // try to find by tag
+	if err != nil {
+		return nil, err
```

Comment:
t his means it's an error if no document is found, is that correct? 

I'd also wrap this error, and combine the query declaration directly into the FindOne() call

Reply:
We seem to have the convention of declaring query separately, which i would prefer to keep. 
This does not error if there is no document found, which is annoying but fits the convention of how FindOne is used.